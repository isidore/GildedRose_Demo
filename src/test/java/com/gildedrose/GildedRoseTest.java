package com.gildedrose;

import com.github.larseckart.tcr.FastTestCommitRevertMainExtension;
import com.github.larseckart.tcr.TestCommitRevertExtension;
import org.approvaltests.Approvals;
import org.approvaltests.combinations.CombinationApprovals;
import org.approvaltests.reporters.AutoApproveReporter;
import org.approvaltests.reporters.UseReporter;
import org.approvaltests.utils.parseinput.ParseInput;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.lambda.utils.Range;

import static org.junit.jupiter.api.Assertions.assertEquals;
//@ExtendWith(TestCommitRevertExtension.class)
@ExtendWith(FastTestCommitRevertMainExtension.class)
public class GildedRoseTest {

    @Test
    public void foo() {
        String[] names = {"foo", "Aged Brie", "Backstage passes to a TAFKAL80ETC concert", "Sulfuras, Hand of Ragnaros"};
        Integer[] qualities = Range.get(-1, 52);
        Integer[] sellIns = Range.get(-1, 20);
        CombinationApprovals.verifyAllCombinations(GildedRoseTest::doStuff, names, qualities, sellIns);
    }

    @Test
    public void conjuredCarrot() {
        Approvals.verify(doStuff("Conjured Carrot", 10, 11));
    }

    @Test
    private static String doStuff(String name, Integer quality, Integer sellIn) {
        Item[] items = new Item[]{new Item(name, sellIn, quality)};
        GildedRose app = new GildedRose(items);
        app.updateQuality();
        String result = app.items[0].toString();
        return result;
    }

}
