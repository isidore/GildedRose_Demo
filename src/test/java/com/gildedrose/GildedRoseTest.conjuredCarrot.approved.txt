Conjured Carrot, 10, 8